e2e/cypress/integration/ad_ldap/saml_ldap_sync_id_attrib_spec.js

```diff
@@ -0,0 +1,185 @@
+
+import {getAdminAccount} from '../../support/env';
+import {getRandomId} from '../../utils';
+
+// assumes that E20 license is uploaded
+// assumes openldap docker available on config default http://localhost:389
+// assumes keycloak docker - uses api to update
+// assumes the CYPRESS_* variables are set (CYPRESS_keycloakBaseUrl / CYPRESS_keycloakAppName)
+// requires {"chromeWebSecurity": false}
+// copy ./mattermost-server/build/docker/keycloak/keycloak.crt -> ./mattermost-server/config/saml-idp.crt
+describe('ad_ldap', () => {
+    const admin = getAdminAccount();
+    const baseUrl = Cypress.config('baseUrl');
+    const loginButtonText = 'SAML';
+
+    const user3 = {
+        username: 'e2etest.three.saml',
+        password: 'Password1',
+        email: 'e2etest.three@mmtest.com',
+        firstname: 'FirstSaml',
+        lastname: 'ThreeSaml',
+        ldapfirstname: 'TestLDAP',
+        ldaplastname: 'ThreeLDAP',
+        keycloakId: '',
+    };
+    const users = {user3};
+
+    const {
+        keycloakBaseUrl,
+        keycloakAppName,
+    } = Cypress.env();
+
+    const idpUrl = `${keycloakBaseUrl}/auth/realms/${keycloakAppName}/protocol/saml`;
+    const idpDescriptorUrl = `${keycloakBaseUrl}/auth/realms/${keycloakAppName}`;
+
+    const newConfig = {
+        SamlSettings: {
+            Enable: true,
+            EnableSyncWithLdap: false,
+            EnableSyncWithLdapIncludeAuth: false,
+            Verify: true,
+            Encrypt: false,
+            SignRequest: false,
+            IdpUrl: idpUrl,
+            IdpDescriptorUrl: idpDescriptorUrl,
+            IdpMetadataUrl: '',
+            ServiceProviderIdentifier: `${Cypress.config('baseUrl')}/login/sso/saml`,
+            AssertionConsumerServiceURL: `${Cypress.config('baseUrl')}/login/sso/saml`,
+            SignatureAlgorithm: 'RSAwithSHA256',
+            CanonicalAlgorithm: 'Canonical1.0',
+            IdpCertificateFile: 'saml-idp.crt',
+            PublicCertificateFile: '',
+            PrivateKeyFile: '',
+            IdAttribute: 'username',
+            GuestAttribute: '',
+            EnableAdminAttribute: false,
+            AdminAttribute: '',
+            FirstNameAttribute: 'firstName',
+            LastNameAttribute: 'lastName',
+            EmailAttribute: 'email',
+            UsernameAttribute: 'username',
+            LoginButtonText: loginButtonText,
+        },
+        LdapSettings: {
+            EnableSync: true,
+            BaseDN: 'ou=e2etest,dc=mm,dc=test,dc=com',
+        },
+    };
+
+    let testSettings;
+
+    before(() => {
+        // * Check if server has license for SAML
+        cy.apiRequireLicenseForFeature('SAML');
+
+        // * Check if server has license for LDAP
+        cy.apiRequireLicenseForFeature('LDAP');
+
+        // # Get certificates status and upload as necessary
+        cy.apiGetSAMLCertificateStatus().then((resp) => {
+            const data = resp.body;
+
+            if (!data.idp_certificate_file) {
```

Comment:
We have an existing tests that uploads `saml-idp.crt`, `saml-public.crt` and `saml-private.key`. I wonder if those should be deleted first to ensure that the `idp_certificate_file` is really for keycloak.

Reply:
I'm always uploading now. There is no need to delete, upload overwrites.